lib/lte_link_control/lte_lc_modem_hooks.c

```diff
@@ -16,11 +16,24 @@ NRF_MODEM_LIB_ON_SHUTDOWN(lte_lc_shutdown_hook, on_modem_shutdown, NULL);
 
 static void on_modem_init(int err, void *ctx)
 {
+	extern const enum lte_lc_system_mode lte_lc_sys_mode;
+	extern const enum lte_lc_system_mode_preference lte_lc_sys_mode_pref;
+
 	if (err) {
 		LOG_ERR("Modem library init error: %d, lte_lc not initialized", err);
 		return;
 	}
 
+	if (!IS_ENABLED(CONFIG_LTE_NETWORK_MODE_DEFAULT)) {
```

Comment:
If `CONFIG_LTE_NETWORK_MODE_DEFAULT` is set, the configured `LTE_MODE_PREFERENCE` does not get applied at all.

The system mode changes look nice (because the implementation is much simpler), but I'm afraid that the functionality may be affected as well. I still need to have a deeper look at it.

Reply:
So the desired behavior when `CONFIG_LTE_NETWORK_MODE_DEFAULT` is set, is that the preference should still be applied? Then all we would need to do is read the system mode and re-apply it with the desired mode preference.

Reply:
Now that I tried it, LTE LC Kconfig doesn't allow configuration where network mode is `CONFIG_LTE_NETWORK_MODE_DEFAULT` and preference is NOT `CONFIG_LTE_MODE_PREFERENCE_AUTO`. So, it looks the code change I asked for isn't needed after all, sorry.

Reply:
Ok, I simplified the commit again